src/_guides/language/coming-from/js-to-dart.md

```diff
@@ -0,0 +1,2835 @@
+curly braces (`{}`) results in creating an empty `Map`.
+To create an empty `Set`, precede the `{}` declaration
+with a type argument or assign `{}` to a variable of type `Set`:
+
+_Dart_
+```dart
+final names = <String>{};
+// Set<String> names = {}; // This works, too.
+// final names = {}; // Creates an empty map, not a set.
+```
+
+The following examples provide an overview of the
+basic actions that you can perform on a Dart `Set`.
+The first example shows how to retrieve a value
+from a `Set` using the index operator:
+
+_Dart_
+```dart
+final fruits = {'apple', 'orange', 'pear'};
+final fruit = fruits[1];
+```
+
+Add a value to the end of the `Set` using the
+`add` method. Use the `addAll` method to add multiple values:
+
+_Dart_
+```dart
+final fruits = {'apple', 'orange', 'pear'};
+fruits.add('peach');
+fruits.addAll(['kiwi', 'mango']);
+```
+
+Use one of the following methods in `Set`
+to remove content from the set:
+
+_Dart_
+```dart
+final fruits = {'apple', 'orange', 'pear'};
+// Remove the value 'pear' from the set.
+fruits.remove('pear');
+// Remove all elements in the supplied list from the set.
+fruits.removeAll(['orange', 'apple']);
+// Removes all elements from the list that match the given predicate.
+fruits.removeWhere((fruit) => fruit.contains('p'));
+```
+
+Use `length` to get the number of values in the `Set`:
+
+_Dart_
+```dart
+final fruits = {'apple', 'orange', 'pear'};
+assert(fruits.length == 2);
+```
+
+Use `isEmpty` to check if the `Set` is empty:
+
+_Dart_
+```dart
+var fruits = <String>{};
+assert(fruits.isEmpty);
+```
+
+Use `isNotEmpty` to check if the `Set` is not empty:
+
+_Dart_
+```dart
+final fruits = {'apple', 'orange', 'pear'};
+assert(fruits.isNotEmpty);
+```
+
+### Maps
+
+The `Map` type in Dart is similar to the `Map` type
+in JavaScript. Both types associate keys with values.
+A key can be any object type, so long as all keys have
+the same type; the same is true for values.
+Each key occurs once at most, but you can use the
+same value multiple times. 
+
+In Dart, the dictionary is based on a hash table,
+which means that keys need to be hashable. In Dart,
+every object contains a unique hash.
+
+{{site.alert.note}}
+  In Dart, the hash value defaults to an instance
+  of an object but you can override it to resemble
+  a data class. For more information,
+  check out the [`hashCode`][] property page.
+{{site.alert.end}}
+
+Here are a couple of simple `Map` examples,
+created using literals:
+
+_Dart_
+```dart
+final gifts = {
+ 'first': 'partridge',
+ 'second': 'turtle doves',
+ 'fifth': 'golden rings'
+};
+
+final nobleGases = {
+ 2: 'helium',
+ 10: 'neon',
+ 18: 'argon',
+};
+```
+
+The following code samples provide an overview of the
+basic actions that you can perform on a Dart `Map`.
+The first example shows how to retrieve a value from
+a `Map` using the key operator:
+
+_Dart_
+```dart
+final gifts = {'first': 'partridge'};
+final gift = gifts['first'];
+```
```

Comment:
I'm not exactly sure how to mention it here, but it seems like it may be helpful to mention how the index (`[]`) operator returns a nullable value.